Remove the Podman preset and all its usages

vyasgun opened the pull request:
**Fixes:** Issue #3970 

**Relates to:** Issue #3970, PR #4012
## Solution/Idea
This PR removes the usages of the `Podman` from the code. This is a follow-up to the PR: https://github.com/crc-org/crc/pull/4012 which added a deprecated warning for the users who chose the `Podman` preset.

## Proposed changes
The changes to remove the usage of Podman preset have been done in two PRs. (https://github.com/crc-org/crc/pull/4012)
- The first change was to print a deprecation warning for users using the preset. 
- In the current PR, the preset itself is being removed from the code.

## Testing
Setting and using `crc` with `Podman` preset should fail.

vyasgun committed:
Podman preset deprecation: Remove all usages of the Podman preset from the code
- remove podman preset specific code
- reject podman as a valid preset value for new machines

anjannath commented on cmd/crc/cmd/version.go:
```diff
@@ -37,15 +37,13 @@ type version struct {
 	Version          string `json:"version"`
 	Commit           string `json:"commit"`
 	OpenshiftVersion string `json:"openshiftVersion"`
-	PodmanVersion    string `json:"podmanVersion"`
```
this should not be removed, the podman version here is also used in the  `openshift` and `microshift` presets

vyasgun replied:
Thanks! Added it back in second commit :)

vyasgun commented:
Checks which set `podman` as preset are failing

anjannath commented:
> Checks which set `podman` as preset are failing

yeah, `integration-crc` and `e2e-crc` are also failing because of `podman` preset related tests, i think you can remove the tests as well in this PR

gbraad commented on .github/workflows/macos-installer.yml:
```diff
@@ -44,11 +44,11 @@ jobs:
         run: sudo installer -pkg out/macos-universal/crc-macos-installer.pkg -target /
-      - name: Set podman preset as part of config
-        run: crc config set preset podman
+      - name: Set microshift preset as part of config
```
Running the microshift preset might result in a very diferent test, so best to let @jsliacan and @adrianriobo chime in on this.

gbraad commented on cmd/crc/cmd/start.go:
```diff
@@ -87,10 +87,6 @@ func runStart(ctx context.Context) (*types.StartResult, error) {
 	}
 
-	if startConfig.Preset == preset.Podman {
-		logging.Warn(preset.PodmanDeprecatedWarning)
```
Do we need to display a removal notice and refer people to use `podman machine` instead?
/cc: @crc-org/crc-team

anjannath replied:
how long will we show this message, is it going to be removed after a few releases? or we'll always show the message when user runs `crc config set podman preset`

another thing is this PR removes `preset.Podman` constant, i don't think we should keep this around to have this check and instead use `Preset("podman")`

gbraad replied:
if you do `crc start` when the preset has already been set to `podman` what is the output now?

anjannath replied:
there are repeated non fatal errors complaining about the preset, both for `setup` and `start` and once we run `crc config unset preset` the errors go away

i think we need to inform the user how to get rid of the errors

```
% crc setup
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO unexpected preset mode podman, using default
ERRO unexpected preset mode podman, using default
INFO Using bundle path /Users/anath/.crc/cache/crc_vfkit_4.14.12_arm64.crcbundle
INFO Checking if running macOS version >= 13.x
INFO Checking if running as non-root
INFO Checking if crc-admin-helper executable is cached
INFO Checking if running on a supported CPU architecture
INFO Checking if crc executable symlink exists
INFO Checking minimum RAM requirements
INFO Check if Podman binary exists in: /Users/anath/.crc/bin/oc
INFO Checking if running emulated on Apple silicon
INFO Checking if vfkit is installed
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if /Users/anath/.crc/cache/crc_vfkit_4.14.12_arm64.crcbundle exists
INFO Checking if old launchd config for tray and/or daemon exists
INFO Checking if crc daemon plist file is present and loaded
INFO Checking SSH port availability
Your system is correctly setup for using CRC. Use 'crc start' to start the instance
ERRO unexpected preset mode podman, using default
```

---

```
% crc start --log-level debug
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
ERRO[0000] unexpected preset mode podman, using default
DEBU CRC version: 2.33.0+93a8fc
DEBU OpenShift version: 4.14.12
DEBU Podman version: 4.4.4
DEBU Running 'crc start'
ERRO unexpected preset mode podman, using default
DEBU Total memory of system is 34359738368 bytes
ERRO unexpected preset mode podman, using default
ERRO unexpected preset mode podman, using default
DEBU No new version available. The latest version is 2.33.0
ERRO unexpected preset mode podman, using default
DEBU Checking file: /Users/anath/.crc/machines/crc/.crc-exist
ERRO unexpected preset mode podman, using default
INFO Using bundle path /Users/anath/.crc/cache/crc_vfkit_4.14.12_arm64.crcbundle
INFO Checking if running macOS version >= 13.x
DEBU kern.osproductversion is: 14.3.1
INFO Checking if running as non-root
INFO Checking if crc-admin-helper executable is cached
DEBU Running '/Users/anath/.crc/bin/crc-admin-helper-darwin --version'
DEBU Found crc-admin-helper-darwin version 0.5.2
DEBU crc-admin-helper executable already cached
INFO Checking if running on a supported CPU architecture
DEBU GOARCH is arm64 GOOS is darwin
INFO Checking if crc executable symlink exists
INFO Checking minimum RAM requirements
DEBU Total memory of system is 34359738368 bytes
INFO Check if Podman binary exists in: /Users/anath/.crc/bin/oc
INFO Checking if running emulated on Apple silicon
INFO Checking if vfkit is installed
DEBU Running '/Users/anath/.crc/bin/vfkit -v'
DEBU Found vfkit version 0.5.0
INFO Checking if old launchd config for tray and/or daemon exists
INFO Checking if crc daemon plist file is present and loaded
DEBU Running 'bash -c launchctl list | grep com.redhat.crc.daemon | awk '{print $1}''
INFO Checking SSH port availability
DEBU Checking file: /Users/anath/.crc/machines/crc/.crc-exist
INFO Loading bundle: crc_vfkit_4.14.12_arm64...
ERRO unexpected preset mode podman, using default
DEBU Using secret from configuration
INFO Creating CRC VM for OpenShift 4.14.12...
```

cfergeau replied:
would be useful to have a removal notice for a while. We can remove it when we remember about it :)

gbraad replied:
I would suggest disallowing starting with this. A removal and alternative use message in that case, right?

vyasgun replied:
I think we can return an error when user tries to start an existing machine with podman preset and modify tests to expect that error

vyasgun committed:
Starting an existing machines with podman preset will return an error

- return deprecation error when the user tries to start an existing machine with podman preset

jsliacan commented on test/integration/podman_test.go:
Maybe worth keeping this test: it verifies podman-remote functionality provided by a podman-preset. However, we can check that functionality for a different preset, microshift requires less resources than openshift, so maybe that's the candidate. I think this sort of modification should work: https://github.com/jsliacan/crc/blob/remove-podman-preset-qe/test/integration/podman_test.go

vyasgun replied:
Thanks! I have made the suggested changes.

jsliacan commented on test/e2e/testsuite/testsuite.go:
```diff
@@ -656,15 +656,9 @@ func FileExistsInCRCHome(fileName string) error {
-	} else {
-		p = preset.OpenShift
-	}
+	var p = preset.OpenShift
```
I think this part was written before microshift preset existed(?) so maybe it's good to use this opportunity to fix this and distinguish between `microshift` and `openshift` presets, while dropping `podman` preset.

jsliacan replied:
There are a few additional places in this file that maybe could be cleaned up at this occasion - if you have time?  Something like in here: https://github.com/jsliacan/crc/pull/17/files

vyasgun replied:
> There are a few additional places in this file that maybe could be cleaned up at this occasion - if you have time? Something like in here: https://github.com/jsliacan/crc/pull/17/files

yes, that makes sense. Will look into it.



> I think this part was written before microshift preset existed(?) so maybe it's good to use this opportunity to fix this and distinguish between `microshift` and `openshift` presets, while dropping `podman` preset.

Maybe @cfergeau @gbraad can chime in if we should default to `microshift`.

gbraad replied:
> if we should default to microshift.

What do you mean with 'default to' ?

vyasgun replied:
Sorry about the confusion for the mix up with 'default to'. What I meant was to replace the Podman preset with Microshift in tests that are currently utilizing the Podman preset.

cfergeau replied:
If we don't have similar tests in the microshift or openshift preset, then yes, doing podman->microshift makes sense

cfergeau replied:
```
	s.Step(`^removing (podman|openshift) bundle from cache succeeds$`,
		RemoveBundleFromCache)
```
If you change `(podman | openshift)` to `openshift` here, maybe you can drop the argument to `RemoveBundleFromCache`?

jsliacan reviewed (commented):
There are also two feature files that could do with some trimming if `podman` preset isn't going to exist. 

- test/e2e/features/minimal.feature
- test/e2e/features/proxy.feature

The diffs that I could think of are in the same example: 
https://github.com/jsliacan/crc/pull/17/files?diff=split&w=0

praveenkumar commented on test/integration/podman_test.go:
```diff
@@ -28,15 +28,24 @@ var _ = Describe("podman preset", Serial, Ordered, Label("podman-preset"), func(
+			if bundlePath == "" {
+				Expect(RunCRCExpectSuccess("setup")).To(ContainSubstring("Your system is correctly setup for using CRC"))
+			} else {
+				Expect(RunCRCExpectSuccess("setup", "-b", bundlePath)).To(ContainSubstring("Your system is correctly setup for using CRC"))
```
You don't need to provide the bundlePath because this is link to OCP and not microshift one. so better to just `setup` and the `start` with pull secret.

vyasgun replied:
Changed

praveenkumar commented:
@vyasgun You need to squash some of commits like 
- https://github.com/crc-org/crc/pull/4062/commits/dc623f37aefea2ea5256b58645f4446220561bf9 
- https://github.com/crc-org/crc/pull/4062/commits/93a8fc84fc86cd1a3783e0626b2221d3aa249f5b 

And 
- https://github.com/crc-org/crc/pull/4062/commits/37b663bd938294d13df0c24c927e132d5ef38a9b
- https://github.com/crc-org/crc/pull/4062/commits/887279e26b3b2056199a3774404b4d37e968697c

And
- https://github.com/crc-org/crc/pull/4062/commits/49be679a945ea24d9e5f8498ca8ab1d21ba77534
- https://github.com/crc-org/crc/pull/4062/commits/6c6ad5d2d0c5b503c87086a1622a7403aeb6ae73

And 
- https://github.com/crc-org/crc/pull/4062/commits/c87073dffe4ed0ca05f0dad2821d644be26e4e35
- https://github.com/crc-org/crc/pull/4062/commits/fe19acf92b3703efc58191502b1ea58a83405454

So that required change are part of single commit and easy for review.

vyasgun commented:
@praveenkumar thanks for the review! The first two commits have been deliberately separated for better clarity on why it is difficult to completely get rid of the using the defined preset. Squashed the others as suggested.

cfergeau commented on pkg/crc/config/settings_test.go:
```diff
@@ -57,17 +57,6 @@ func TestCPUsValidate(t *testing.T) {
 	}, cfg.Get(CPUs))
-
-	_, err = cfg.Set(Preset, crcpreset.Podman)
-	require.NoError(t, err)
```
The comment before the test says:
```
// Check that with the default preset, we cannot set less CPUs than the defaultCPUs
// but that it is allowed with a different preset with less requirements (podman)
```

If you remove the end of the test, what it does no longer matches its description. You can replace `Podman` with `Microshift` here and keep the rest of the code.

cfergeau commented on pkg/crc/machine/bundle/metadata.go:
```diff
@@ -176,12 +176,7 @@ func (bundle *CrcBundleInfo) GetPodmanVersion() string {
-	default:
-		return bundle.GetOpenshiftVersion()
-	}
+	return bundle.GetOpenshiftVersion()
```
A followup commit could remove `GetOpenshiftVersion()` or make it private since we now have 2 methods with different names which return the same thing.

vyasgun replied:
Good point. Removed the redundant function.

cfergeau commented on pkg/crc/machine/bundle/metadata.go:
```diff
@@ -205,22 +200,16 @@ func (bundle *CrcBundleInfo) IsMicroshift() bool {
-			bundle.GetOcPath(),
-			bundle.GetKubeConfigPath()}...)
-	}
+	files = append(files, []string{
```
You can move these directly into the initialization of `files` just above:
```
	files := []string{
```

vyasgun replied:
thanks! I didn't catch that.

cfergeau commented on pkg/crc/machine/bundle/repository.go:
```diff
@@ -55,17 +55,16 @@ func (repo *Repository) Get(bundleName string) (*CrcBundleInfo, error) {
-		if bundleInfo.GetAPIHostname() != fmt.Sprintf("api%s", constants.ClusterDomain) {
-			return nil, fmt.Errorf("unexpected bundle, it must have %s base domain", constants.ClusterDomain)
-		}
+	// TODO: update this logic after major release of bundle like 4.14
```
could be a good time to check if we can finally change this?

vyasgun replied:
I'm not sure what exactly needs to be done here -- can you or @praveenkumar help me with some pointers?

cfergeau replied:
This resonates with what I was saying in https://github.com/crc-org/crc/pull/4062#issuecomment-2027375311

If you look at `tig blame` (or `git blame`) for this code https://github.com/crc-org/crc/blame/main/pkg/crc/machine/bundle/repository.go#L60 you get this commit https://github.com/crc-org/crc/commit/d72f292a0d9e8bbbf94aa4e041aa38e32183212f which has more details in its commit log.

Still not fully sure what needs to be changed, maybe in the line `if !crcstrings.Contains([]string{constants.AppsDomain, constants.MicroShiftAppDomain}, fmt.Sprintf(".%s", bundleInfo.ClusterInfo.AppsDomain)) {` do:
- if openshift bundle, check for `constants.AppsDomain` and not `constants.MicroshiftAppsDomain`
- if microshift bundle, only check for `constants.MicroshiftAppsDomain`

The check would be similar to the code before this change:
```
if fmt.Sprintf(".%s", bundleInfo.ClusterInfo.AppsDomain) != constants.AppsDomain {
			return nil, fmt.Errorf("unexpected bundle, it must have %s apps domain", constants.AppsDomain)
}
```

No hurries for this though, this can be done in a separate/additional PR, let's not block the PR on this.

gbraad replied:
> let's not block the PR on this.

I agree.

vyasgun replied:
okay, I'll create a new PR for this

cfergeau commented on pkg/crc/machine/start.go:
```diff
@@ -481,34 +474,6 @@ func (client *client) Start(ctx context.Context, startConfig types.StartConfig)
-		if err := configurePodmanProxy(sshRunner, proxyConfig); err != nil {
-			return nil, errors.Wrap(err, "Failed to configure proxy for podman")
-		}
-		if err := dns.AddPodmanHosts(instanceIP); err != nil {
```
You can remove this function from the `dns` package, it's no longer used after your changes.

cfergeau commented:
> The first two commits have been deliberately separated for better clarity on why it is difficult to completely get rid of the using the defined preset.

I had this comment in mind when reviewing the first 2 commits, and I must say that looking at the 2nd commit, all I see is that you readd code that you removed in the first commit. It's not obvious at all to me what the problem was with the removal, why you had to readd this code, ... I'm sure there is a good reason, but the split in 2 commits is not helping me to understand it :) The commit log is there to document the reasons making the changes needed, one-line commit logs should be the exception, not the norm, see https://www.berrange.com/tags/commit-message/ for example for some lengthy explanation / recommendations.

cfergeau commented on pkg/crc/machine/bundle/metadata.go:
```diff
@@ -200,6 +200,10 @@ func (bundle *CrcBundleInfo) IsMicroshift() bool {
+func (bundle *CrcBundleInfo) IsPodman() bool {
+	return bundle.GetBundleType() == crcPreset.Podman
+}
+
```
You could keep some of these in the initial commit if you readd these now?

cfergeau replied:
Couldn't this be kept in the first commit?

cfergeau reviewed (commented):
Out of curiosity, what happens on upgrade if you do this?
```
$ crc.old config set preset podman
$ crc.old start
$ crc.old stop
$ crc.new start
```
(`crc.old` has podman preset support, `crc.new` no longer has it).

vyasgun commented:
@cfergeau Thanks a lot for the detailed review and comments! In my head, the separation of commits made sense but I don't think it's reflecting that way 😓  I have addressed most of the comments - hope the changes are okay now.

I have tested with old and new crc and if there was an existing machine with podman preset, the new crc will throw an error and ask the user to move to podman machine. [Code for this.](https://github.com/crc-org/crc/pull/4062/files#diff-0ee0833dbc0dcbd3f2822c9578f871dfb4a3662f3e12cc4d6eef36fac6524828L90)

cfergeau commented on pkg/crc/config/settings_test.go:
```diff
@@ -57,17 +57,6 @@ func TestCPUsValidate(t *testing.T) {
 		IsSecret:  false,
 	}, cfg.Get(CPUs))
-
-	_, err = cfg.Set(Preset, crcpreset.Podman)
```
This part of the code needs to be kept with `crcpreset.Podman` -> `crcpreset.Microshift`

cfergeau commented on pkg/crc/preflight/preflight_darwin_test.go:
```diff
@@ -13,13 +13,13 @@ import (
 	cfg := config.New(config.NewEmptyInMemoryStorage(), config.NewEmptyInMemorySecretStorage())
 	RegisterSettings(cfg)
-	assert.Len(t, cfg.AllConfigs(), 13)
+	assert.Len(t, cfg.AllConfigs(), 14)
```
These changes should no longer be needed after fe2a9176676
Can you rebase your changes on top of `origin/main`?

cfergeau commented on .github/workflows/macos-installer.yml:
```diff
@@ -44,11 +44,7 @@ jobs:
-      - name: Wait 10 sec for the daemon to serve
-        run: sleep 10
-      - name: Run crc with podman preset
-        run: crc start
```
The commit log is
```
 modified github workflow

removing crc start from github test
```

This describes what the commit is doing, but this does not explain the reason why you want to make this change. My guess is that this is required as we cannot reliably run `crc start` for the microshift preset in github actions because the runners are not powerful enough, but this is just a guess, maybe you have a very different reason for this change, and I cannot know that from just the commit log and the diff. Can you explain this in the git log?

vyasgun replied:
The main reason is that with podman preset, authentication was not required from user's end.  As per previous discussions, `crc start` with `microshift` preset is not necessary as a part of the github workflow. @praveenkumar is that correct?

cfergeau replied:
I'm still not clear what "authentication" refers to here? Does this refer to the pull secret which is needed for `crc start` with microshift? And since we don't have a valid pull secret in gh actions, we can't run crc start?

vyasgun replied:
Yes, that's what I mean. The pull secret requirement means new steps have to be added in the github actions. This step was added in [this commit.](https://github.com/vyasgun/crc/commit/a4c61d9e9c2797aa6487c8f219bef26c5d045d58) I had a discussion with @praveenkumar about this and it seemed like a better idea to remove the `start` command.

cfergeau reviewed (commented):
Still a few comments, but overall looks fine!

gbraad commented:
Can we run some tests to ensure the flow works for a regular OpenShift deployment from PD? Followed by a situation where the `podman` preset was chosen in an old version, but running with this. Please share output messages to ensure user-facing information is correct for `crc config`, `crc start` and `crc delete`.

vyasgun commented:
The following shows the effect of the starting an older machine with podman preset with the updated crc (It will throw an error and redirect user to podman machine)
```
gvyas-mac:~ gvyas$ crc.old config view
- consent-telemetry                     : no
- disable-update-check                  : true
- preset                                : podman

gvyas-mac:~ gvyas$ crc.old setup
INFO Using bundle path /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle
INFO Checking if running macOS version >= 13.x
INFO Checking if running as non-root
INFO Checking if crc-admin-helper executable is cached
INFO Checking if running on a supported CPU architecture
INFO Checking if crc executable symlink exists
INFO Creating symlink for crc executable
INFO Checking minimum RAM requirements
INFO Check if Podman binary exists in: /Users/gvyas/.crc/bin/oc
INFO Checking if running emulated on Apple silicon
INFO Checking if vfkit is installed
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle exists
INFO Checking if old launchd config for tray and/or daemon exists
INFO Checking if crc daemon plist file is present and loaded
INFO Adding crc daemon plist file and loading it
INFO Checking SSH port availability
Your system is correctly setup for using CRC. Use 'crc start' to start the instance

gvyas-mac:~ gvyas$ crc.old start
WARN The Podman preset is deprecated and will be removed in a future release. Consider rather using a Podman Machine managed by Podman Desktop: https://podman-desktop.io
INFO Using bundle path /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle
INFO Checking if running macOS version >= 13.x
INFO Checking if running as non-root
INFO Checking if crc-admin-helper executable is cached
INFO Checking if running on a supported CPU architecture
INFO Checking if crc executable symlink exists
INFO Checking minimum RAM requirements
INFO Check if Podman binary exists in: /Users/gvyas/.crc/bin/oc
INFO Checking if running emulated on Apple silicon
INFO Checking if vfkit is installed
INFO Checking if old launchd config for tray and/or daemon exists
INFO Checking if crc daemon plist file is present and loaded
INFO Checking SSH port availability
INFO Loading bundle: crc_podman_vfkit_4.4.4_amd64...
INFO Creating CRC VM for Podman 4.4.4...
INFO Generating new SSH key pair...
INFO Starting CRC VM for podman 4.4.4...
INFO CRC instance is running with IP 127.0.0.1
INFO CRC VM is running
INFO Updating authorized keys...
INFO Configuring shared directories
INFO Adding new bearer token for cockpit webconsole
podman runtime is now running.

Use the 'podman' command line interface:
  $ eval $(crc podman-env)
  $ podman COMMAND

gvyas-mac:~ gvyas$ crc.old stop
INFO Updating kernel args...
INFO Stopping the instance, this may take a few minutes...
Stopped the instance
```

```
gvyas-mac:~ gvyas$ crc.new config view
- consent-telemetry                     : no
- disable-update-check                  : true
- preset                                : podman

gvyas-mac:~ gvyas$ crc.new start
INFO Using bundle path /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle
INFO Checking if running macOS version >= 13.x
INFO Checking if running as non-root
INFO Checking if crc-admin-helper executable is cached
INFO Checking if running on a supported CPU architecture
INFO Checking if crc executable symlink exists
WARN Preflight checks failed during `crc start`, please try to run `crc setup` first in case you haven't done so yet
/Users/gvyas/.crc/bin/crc points to /usr/local/bin/crc.old, not to /usr/local/bin/crc.new

gvyas-mac:~ gvyas$ crc.new setup
INFO Using bundle path /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle
INFO Checking if running macOS version >= 13.x
INFO Checking if running as non-root
INFO Checking if crc-admin-helper executable is cached
INFO Checking if running on a supported CPU architecture
INFO Checking if crc executable symlink exists
INFO Creating symlink for crc executable
INFO Checking minimum RAM requirements
INFO Check if Podman binary exists in: /Users/gvyas/.crc/bin/oc
INFO Checking if running emulated on Apple silicon
INFO Checking if vfkit is installed
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle exists
INFO Getting bundle for the CRC executable
INFO Downloading bundle: /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle...
Getting image source signatures
Copying blob c8fcbc8cecc6 done   |
Copying config 60347c49ff done   |
Writing manifest to image destination
INFO Extracting the image bundle layer...
crc_podman_vfkit_4.4.4_amd64.crcbundle:  1.05 GiB / 1.05 GiB [-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
INFO Verifying the bundle signature...
INFO Uncompressing /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle
crc-podman.img:  31.00 GiB / 31.00 GiB [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00%
unexpected bundle, it must have .apps-crc.testing or .apps.crc.testing apps domain

gvyas-mac:~ gvyas$ crc.new start
INFO Using bundle path /Users/gvyas/.crc/cache/crc_podman_vfkit_4.4.4_amd64.crcbundle
INFO Checking if running macOS version >= 13.x
INFO Checking if running as non-root
INFO Checking if crc-admin-helper executable is cached
INFO Checking if running on a supported CPU architecture
INFO Checking if crc executable symlink exists
INFO Checking minimum RAM requirements
INFO Check if Podman binary exists in: /Users/gvyas/.crc/bin/oc
INFO Checking if running emulated on Apple silicon
INFO Checking if vfkit is installed
INFO Checking if old launchd config for tray and/or daemon exists
INFO Checking if crc daemon plist file is present and loaded
INFO Checking SSH port availability
The Podman preset is deprecated and will be removed in a future release. Consider rather using a Podman Machine managed by Podman Desktop: https://podman-desktop.io
```

vyasgun commented:
The error is not being shown as expected in Podman Desktop:

1. <img width="522" alt="image" src="https://github.com/crc-org/crc/assets/87057902/ea3b8930-c609-4c7b-98d0-92e3a2b50bff">
2. <img width="523" alt="image" src="https://github.com/crc-org/crc/assets/87057902/14ad2808-4444-4004-87a8-81b168c5ad28">
3. <img width="273" alt="image" src="https://github.com/crc-org/crc/assets/87057902/c4303f38-8c98-4136-aa00-9f2e1a034ad8">
4. <img width="521" alt="image" src="https://github.com/crc-org/crc/assets/87057902/b39c00f2-bd40-4a11-b809-440d89c0840a">

vyasgun commented:
/retest

cfergeau commented on pkg/crc/config/settings_test.go:
```diff
@@ -57,6 +57,17 @@ func TestCPUsValidate(t *testing.T) {
+		Invalid:   false,
+		IsDefault: false,
+		IsSecret:  false,
+	}, cfg.Get(CPUs))
```
This change belongs in the previous commit

cfergeau reviewed (commented):
Overall looks good to me, still one commit log to expand a bit, and a change to move from one commit to another one

vyasgun committed:
Test changes for podman preset removal

- change preset to microshift for podman bundle tests
- change preset to microshift in github workflow
- remove crc start from github workflow to avoid auth error

gbraad commented:
> The error is not being shown as expected in Podman Desktop:

Let's create new issues for this. The preset was never possible to set from the extension, so we need to decide how much this matters/should be resolved. It does not impact this PR.

gbraad commented:
/lgtm

cfergeau reviewed (approved):
Just one pending question, but the code is fine!

gbraad commented:
> Just one pending question

does this impede/prevent merging? Might be great to indicate which question you mean.

cfergeau commented:
> > Just one pending question
> 
> does this impede/prevent merging? Might be great to indicate which question you mean.

This was the only other comment I added yesterday, and Gunjan answered it so she must have found it even if I can't find it now :-/

cfergeau commented:
> > > Just one pending question
> > 
> > 
> > does this impede/prevent merging? Might be great to indicate which question you mean.
> 
> This was the only other comment I added yesterday, and Gunjan answered it so she must have found it even if I can't find it now :-/

https://github.com/crc-org/crc/pull/4062#discussion_r1570850298

The commit log can't be changed after merging, so if we want to make it clearer it's now or never. I don't feel too strongly about it, if Gunjan did not change the commit log, then fine with me.
